### PR TITLE
changes add knowledge dialog 

### PR DESCRIPTION
**Pull request by plinionr3**
A slider in the dialog to choose the knowledge weight.

### CODE REVIEW

File: frontend/src/views/RoleView.vue [hunk 1]
```diff
@@ -20,6 +20,10 @@
         <ElButton type="primary" @click="showDialogAddNewKnowledge">Add new Knowledge</ElButton>
         <ElDialog v-model="createKnowledgeDialogFormVisible" title="New Knowledge" width="500">
           <ElInput v-model="newKnowledgeName" placeholder="Knowledge name" :clearable="false" />
+          <div class="slider-demo-block">
```
Comment (vitorlofonseca) [inline frontend/src/views/RoleView.vue:23]: Name your classes right. This is not a slider demo

File: frontend/src/views/RoleView.vue [hunk 2]
```diff
@@ -49,13 +54,15 @@ const onEditRow = (row: Knowledge) => {
 }
 
 const roleStore = useRolesStore()
+const knowledgesStore = useKnowledgeStore()
 const createKnowledgeDialogFormVisible = ref(false)
 const newKnowledgeName = ref('')
-const roleId = ref<string>('1')
+const roleId = ref<number>(1)
```
Comment (vitorlofonseca) [inline frontend/src/views/RoleView.vue:60]: You don't need to create a variable just for this. It's just a mock

File: frontend/src/views/RoleView.vue [hunk 3]
```diff
@@ -49,13 +54,15 @@ const onEditRow = (row: Knowledge) => {
 }
 
 const roleStore = useRolesStore()
+const knowledgesStore = useKnowledgeStore()
 const createKnowledgeDialogFormVisible = ref(false)
 const newKnowledgeName = ref('')
```
Comment (vitorlofonseca) [inline frontend/src/views/RoleView.vue:59]: Can you put the name and the weight both in the same object?

**Review by fromgonie — COMMENTED** [review]
Good Job!

**Review by fromgonie — APPROVED** [review]
Now we speaking the same language!

File: frontend/src/views/RoleView.vue [hunk 4]
```diff
@@ -19,7 +19,11 @@
         <br />
         <ElButton type="primary" @click="showDialogAddNewKnowledge">Add new Knowledge</ElButton>
         <ElDialog v-model="createKnowledgeDialogFormVisible" title="New Knowledge" width="500">
-          <ElInput v-model="newKnowledgeName" placeholder="Knowledge name" :clearable="false" />
+          <ElInput v-model="knowledge.name" placeholder="Knowledge name" :clearable="false" />
+          <div class="SliderBlock">
```
Comment (vitorlofonseca) [inline frontend/src/views/RoleView.vue:23]: You don't have to describe what your class is doing (Block). Your class should just contain the name of your component

Read this to have more info: https://getbem.com/

File: frontend/src/views/RoleView.vue [hunk 5]
```diff
@@ -19,7 +19,11 @@
...
+            <span class="demonstration">Weight</span>
```
Comment (vitorlofonseca) [inline frontend/src/views/RoleView.vue:24]: Keep consistency in class naming (CapitalCasing)

File: frontend/src/views/RoleView.vue [hunk 6]
```diff
@@ -102,4 +107,26 @@ const createKnowledge = async () => {
   margin: auto;
   text-align: left;
 }
+.SliderBlock {
+  max-width: 600px;
+  display: flex;
+  align-items: center;
+}
+.SliderBlock .ElSlider {
```
Comment (vitorlofonseca) [inline frontend/src/views/RoleView.vue:115]: Please, put the ElSlider styles in the components.scss file

**Review by vascolusitan — APPROVED** [review]

File: frontend/src/views/RoleView.vue [hunk 7]
```diff
@@ -66,24 +71,24 @@ const hideDialog = () => {
...
     return
   }
   try {
-    /*await knowledgesStore.saveKnowledge(
-      name: newKnowledgeName.value,
-      roleId: '1',
-      levelId: '1',
-      weight: '1'
-    )*/
+    await knowledgesStore.saveKnowledge({
```
Comment (vitorlofonseca) [inline frontend/src/views/RoleView.vue:79]: Why not pass the knowledge here? Like this
```suggestion
    await knowledgesStore.saveKnowledge(knowledge)
```

**Review by vitorlofonseca — APPROVED** [review]

**Pull request merged**